File: Scala/src/lab4/ColorTurtle.scala
package lab4

import java.awt.Color;
import se.lth.cs.window.SimpleWindow;

class ColorTurtle ( override val wc: SimpleWindow, 
    override val xc: Int, override val yc: Int, 
    val col: Color) extends Turtle(wc,xc,yc) { 
    var color: Color = col
    
    override def forward(i: Int) {
      w.setLineColor(color)
      super.forward(i)
    }
    
    def setColor(col: Color) = color = col
}

File: inl1/src/lab5/ColorTurtle.java
package lab5;

import java.awt.Color;
import lab4.Turtle;
import se.lth.cs.window.SimpleWindow;

public class ColorTurtle extends Turtle {
	private Color color;

	public ColorTurtle(SimpleWindow w, int x, int y, Color color) {
		super(w, x, y);
		this.color = color;
	}

	public void forward(int n) {
		Color savedColor = w.getLineColor();
		w.setLineColor(color);
		super.forward(n);
		w.setLineColor(savedColor);
	}

	public void setColor(Color color) {
		this.color = color;
	}
}